.github/workflows/e2e.yml

```diff
@@ -0,0 +1,67 @@
+name: End to End Tests
+on: 
+  workflow_dispatch:
+    inputs:
+      automationBranch:
+        description: 'Set the branch to use for automation tests'
+        required: false
+        default: 'develop'
+        type: string
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [macos-latest]
+        vscodeVersion:
+          - stable
+    runs-on: ${{ matrix.os }}
+    steps:
+        - name: Checkout
+          uses: actions/checkout@v2
+          with:
+            path: ./salesforcedx-vscode
+            ref: ${{ github.event.ref }}
+        - uses: actions/setup-node@v3
+          with:
+            node-version-file: 'salesforcedx-vscode/.nvmrc' 
+            cache: npm
+            cache-dependency-path: |
+              salesforcedx-vscode/package-lock.json
+              salesforcedx-vscode-automation-tests/package-lock.json
+        - name: Clone automation tests
+          uses: actions/checkout@v2
+          with:
+            repository: forcedotcom/salesforcedx-vscode-automation-tests
+            path: salesforcedx-vscode-automation-tests
+            ref: ${{ inputs.automationBranch }}
+        - name: Install Extension Dependencies
+          run: |
+            npm install
+          working-directory: salesforcedx-vscode
+        - name: Install Test Dependencies
+          run: |
+            npm install
+          working-directory: salesforcedx-vscode-automation-tests
+        - name: Install the SFDX CLI
+          run: npm install -g sfdx-cli
+        - name: Verify CLI
+          shell: bash
+          run: |
+            set -e
+            sfdx version
+            SFDX_CLI_VERSION=$(sfdx version)
+            if [[ ((`echo $SFDX_CLI_VERSION | grep -c "sfdx-cli/"` > 0))]]
+            then
+              echo "sfdx-cli installed -" $SFDX_CLI_VERSION
+            else
+              echo "The sfdx-cli installation could not be verified"
+              exit 1
+            fi
+        - name: Run headless test
+          uses: GabrielBB/xvfb-action@v1
```

Comment:
did we ever get this 3pp approved?

Reply:
not yet - we can bind to a specific commit hash, or try and run it without?

Reply:
and by "without" I mean I've seen some ugly code that lets us run tests in a headless chrome environment, without using an action specifically :)

Reply:
ohhh better yet, that repo actually recommends using the Open Source Community Action now umbrella'd under Github: https://github.com/coactions + https://github.com/coactions/setup-xvfb . I'll still pin at a commit hash, but this seems a better bet?

Reply:
Looks like the new action works too: https://github.com/randi274/salesforcedx-vscode-fork/actions/runs/4756852368/workflow